enterprise/glassfish.javaee/src/org/netbeans/modules/glassfish/javaee/RunTimeDDCatalog.java

```diff
@@ -239,7 +239,7 @@ public static synchronized RunTimeDDCatalog getEE6RunTimeDDCatalog(){
      * @return null if cannot proceed, try later.
      */
     @Override
-    public Iterator getPublicIDs() {
+    public Iterator<String> getPublicIDs() {
```

Comment:
Wouldn't it make sense to also fix the interface `CatalogReader` that is implemented here? The documentation indicates, that this is right, but I would make it explicit in the code:

> Get String iterator representing all public IDs registered in catalog

This affects all `CatalogReader` implementations.

Reply:
I agree. The `CatalogReader` interface needs updating.

Fixed..